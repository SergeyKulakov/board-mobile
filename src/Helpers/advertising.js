export const getDataWithAds = data => {
  const result = []

  data.forEach((el, key) => {
    const index = key + 1
    if (data[key + 1]) {
      if (index % 5 === 0) {
        result.push(el)
        result.push('ads')
      } else if (index === 3 || index % 7 === 0) {
        result.push(el)
        result.push('sponsor')
      } else result.push(el)
    } else result.push(el)
  })

  return result
}
