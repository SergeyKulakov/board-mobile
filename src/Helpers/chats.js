import moment from 'moment'

export const getMessageTime = date => {
  const TODAY = moment()
  const YESTERDAY = moment().date(TODAY.get('date') - 1)

  const data = Number(date)

  if (moment(data).isSame(TODAY, 'day')) return moment(data).format('HH:mm')
  if (moment(data).isSame(YESTERDAY, 'day')) return 'Yesterday'
  return moment(data).format('DD MMM YYYY')
}
