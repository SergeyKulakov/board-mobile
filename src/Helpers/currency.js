import data from 'Constants/country-currency-codes'

export const getSearchData = string => {
  if (!string || typeof string !== 'string') return data

  const value = string.toLowerCase()

  return data.filter(
    el =>
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.code.toLowerCase().indexOf(value) !== -1,
  )
}

export default data
