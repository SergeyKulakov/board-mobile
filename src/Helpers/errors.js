import _ from 'lodash'
import i18n from 'I18N'

export const getError = (error, path = 'apiErrors') => {
  const message = _.get(error, 'payload.message')
  const code = _.get(error, 'payload.code')

  if (
    error.payload === 'Internal server error' ||
    message === 'Internal server error' ||
    message === 'Internal Server Error' ||
    code === 'internal'
  )
    return i18n.t('apiErrors.InternalServerError')

  if (error.message === 'Unauthorized' || error.payload === 'Unauthorized') {
    return i18n.t('apiErrors.unauthorizedError')
  }

  if (!code) return `${message || error.payload}`

  return i18n.t(`${path}.${code}`)
}
