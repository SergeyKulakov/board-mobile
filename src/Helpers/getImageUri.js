import { BACKET } from 'react-native-dotenv'
import ProfileImagesManager from 'Services/Images/ProfileImagesManager'
import { profileImageTypes } from 'Services/Constants/profileImages.constants'
import _ from 'lodash'
import isUrl from 'is-url'

const s3 = 'https://s3.amazonaws.com'
const backet = BACKET
const serviceFolder = 'service-icons'
const jobsFolder = 'job-pics'

export const getServiceImage = fileName => {
  const iconName = fileName.replace(/\s/g, '%20')

  return `${s3}/${backet}/${serviceFolder}/${iconName}`
}

export const getJobImage = (id, fileName) =>
  `${s3}/${backet}/${jobsFolder}/${id}/${fileName}`

export const getProfileImage = ({ type, src, userId }) => {
  let source = src

  if (_.isString(src)) {
    if (/(data:image\/.*?;base64),/g.exec(src) || src.startsWith('http')) {
      source = src
    } else {
      source = ProfileImagesManager.getSignedURLForProfileImage(
        type,
        src,
        userId,
      )
    }
  } else if (_.isObject(src)) {
    source = null
  }

  return source
}

export const getProfileAvatar = (avatarURL, userId) => {
  if (isUrl(avatarURL)) return avatarURL

  if (_.isString(avatarURL)) {
    return getProfileImage({
      type: profileImageTypes.avatar,
      src: avatarURL,
      userId,
    })
  }

  return null
}
