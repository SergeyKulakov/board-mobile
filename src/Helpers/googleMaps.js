import _ from 'lodash'
import memoize from 'memoize-one'

const googleMaps = 'https://maps.googleapis.com/maps/api/staticmap'
const googleGeocoding = 'https://maps.googleapis.com/maps/api/geocode'
const params = {
  maptype: 'roadmap',
  size: '600x300',
  zoom: '15',
  markers: [
    'color:blue|label:S|40.702147,-74.015794', // example
  ],
  key: 'AIzaSyAOrI9daMiZzf1t2oPscaFD4FpPod4yTR8',
  center: 'BrooklynBridge,New York,NY', // example
}

export const getMapImage = (geolocation = {}) => {
  const { lat, lon } = geolocation
  if (lat && lon) {
    return `${googleMaps}?center=${lat},${lon}&zoom=${params.zoom}&size=${
      params.size
    }&maptype=${
      params.maptype
    }&markers=color:red%7Clabel:S%7C${lat},${lon}&key=${params.key}`
  }

  return `https://maps.googleapis.com/maps/api/staticmap?center=BrooklynBridge,New%20York,NY&zoom=0&size=600x300&maptype=roadmap&key=${
    params.key
  }`
}

export const getLocatinByGeocoding = (lat, lon) => {
  return `${googleGeocoding}/json?latlng=${lat},${lon}&key=${params.key}`
}

export const getGeocodeObg = memoize(geocode => {
  if (_.isString(geocode) && !_.isEmpty(geocode)) {
    const geodata = geocode.split('/')
    return {
      lat: Number(geodata[0]),
      lon: Number(geodata[1]),
    }
  }

  return {}
})
