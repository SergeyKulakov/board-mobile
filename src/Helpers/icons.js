import colors from 'Themes/Colors'
import _ from 'lodash'

export const getNavigationIconStyle = (icon, allSize = 20) => {
  const defaultProps = {
    size: allSize,
    color: colors.blue,
  }

  if (_.isString(icon) || !icon) {
    return {
      name: icon,
      ...defaultProps,
    }
  }

  return {
    type: icon.type,
    name: icon.name,
    color: icon.color || defaultProps.color,
    size: icon.size || defaultProps.size,
  }
}
