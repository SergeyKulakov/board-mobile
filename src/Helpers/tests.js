export const getTestID = value => `[testID="${value}"]`

export const serverErrorObg = {
  error: {
    payload: {
      message: 'message',
      code: 'code',
    },
  },
}
