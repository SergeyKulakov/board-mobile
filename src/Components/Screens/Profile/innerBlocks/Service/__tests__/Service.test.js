import React from 'react'
import { shallow } from 'enzyme'
import { getTestID } from 'Helpers/tests'
import 'jest-styled-components'

import { Service as Component } from '../index'

describe('Profile.Service', () => {
  const defaultProps = {
    text: {
      servicesLengthError: 'servicesLengthError',
    },
    values: {
      services: [],
    },
    setFieldValue: () => null,
    isPremium: false,
    onAddServices: () => null,
    onEditServices: () => null,
    handleBlur: () => null,
    toast: () => null,
    services: [],
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('service actions', () => {
    describe('edit', () => {
      it("shouldn't call edit by default", () => {
        const props = {
          ...defaultProps,
          onEditServices: jest.fn(),
        }

        shallow(<Component {...props} />)

        expect(props.onEditServices).not.toHaveBeenCalled()
      })

      it('shouldn\t call edit without arguments', () => {
        const props = {
          ...defaultProps,
          onEditServices: jest.fn(),
        }

        const wrapper = shallow(<Component {...props} />)

        wrapper.instance().handleEdit()

        expect(props.onEditServices).not.toHaveBeenCalled()
      })

      it('with category', () => {
        const props = {
          ...defaultProps,
          onEditServices: jest.fn(),
        }

        const wrapper = shallow(<Component {...props} />)

        const args = {
          _id: 0,
          categoryId: 1,
        }
        wrapper.instance().handleEdit(args)

        expect(props.onEditServices).toHaveBeenCalledTimes(1)
        expect(props.onEditServices).toHaveBeenCalledWith({
          serviceId: args._id,
          categoryId: args.categoryId,
          step: 2,
        })
      })

      it('without category', () => {
        const props = {
          ...defaultProps,
          onEditServices: jest.fn(),
        }

        const wrapper = shallow(<Component {...props} />)

        const args = {
          _id: 0,
        }
        wrapper.instance().handleEdit(args)

        expect(props.onEditServices).toHaveBeenCalledTimes(1)
        expect(props.onEditServices).toHaveBeenCalledWith({
          categoryId: args._id,
          step: 1,
        })
      })
    })

    it('delete', () => {
      const props = {
        ...defaultProps,
        setFieldValue: jest.fn(),
        values: {
          ...defaultProps.values,
          services: [
            {
              _id: 0,
            },
          ],
        },
      }

      const wrapper = shallow(<Component {...props} />)

      const args = 0
      wrapper.instance().handleDelete(args)

      expect(props.setFieldValue).toHaveBeenCalledTimes(1)
      expect(props.setFieldValue).toHaveBeenCalledWith('services', [])
    })

    describe('add', () => {
      it("shouldn't call add by default", () => {
        const props = {
          ...defaultProps,
          onAddServices: jest.fn(),
        }

        shallow(<Component {...props} />)

        expect(props.onAddServices).not.toHaveBeenCalled()
      })

      it('with premium', () => {
        const props = {
          ...defaultProps,
          onAddServices: jest.fn(),
          isPremium: true,
          values: {
            ...defaultProps.values,
            services: Array(3),
          },
          toast: jest.fn(),
        }

        const wrapper = shallow(<Component {...props} />)

        for (let i = 0; i < 10; i++) {
          wrapper.instance().handleAddService()
          expect(props.onAddServices).toHaveBeenCalledTimes(i + 1)
          expect(props.toast).not.toHaveBeenCalled()
        }
      })

      it('without premium', () => {
        const props = {
          ...defaultProps,
          onAddServices: jest.fn(),
          isPremium: false,
          values: {
            ...defaultProps.values,
            services: [],
          },
          toast: jest.fn(),
        }

        const wrapper = shallow(<Component {...props} />)

        for (let i = 0; i < 10; i++) {
          wrapper.instance().handleAddService()
          if (i <= 3) {
            wrapper.setProps({
              values: {
                services: [...wrapper.instance().props.values.services, 1],
              },
            })
          }
        }

        expect(props.onAddServices).toHaveBeenCalledTimes(3)
        expect(props.toast).toHaveBeenCalledTimes(7)
      })
    })

    it('toggle active status switch', () => {
      const props = {
        ...defaultProps,
        values: {
          services: [
            {
              _id: 0,
              status: 'active',
            },
            {
              _id: 1,
              status: 'inactive',
            },
          ],
        },
        setFieldValue: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      const args = 0

      wrapper.instance().handleToggleActive(args)

      expect(props.setFieldValue).toHaveBeenCalledTimes(1)
      expect(props.setFieldValue).toHaveBeenCalledWith('services', [
        {
          _id: 0,
          status: 'inactive',
        },
        {
          _id: 1,
          status: 'inactive',
        },
      ])

      wrapper.setProps({
        values: {
          services: [
            {
              _id: 0,
              status: 'inactive',
            },
            {
              _id: 1,
              status: 'inactive',
            },
          ],
        },
      })

      const args2 = 1

      wrapper.instance().handleToggleActive(args2)

      expect(props.setFieldValue).toHaveBeenCalledTimes(2)
      expect(props.setFieldValue).toHaveBeenCalledWith('services', [
        {
          _id: 0,
          status: 'inactive',
        },
        {
          _id: 1,
          status: 'active',
        },
      ])
    })
  })

  it('change about input field', () => {
    const props = {
      ...defaultProps,
      setFieldValue: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    const args = 'a'

    wrapper
      .find(getTestID`aboutInput`)
      .props()
      .onChange(args)

    expect(props.setFieldValue).toHaveBeenCalledTimes(1)
    expect(props.setFieldValue).toHaveBeenCalledWith('about', args)
  })

  it('render service item', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    const args = {
      item: {
        _id: 0,
        title: 'title',
        status: 'active',
      },
    }

    wrapper.instance().handleEdit = jest.fn()
    wrapper.instance().handleDelete = jest.fn()
    wrapper.instance().handleOpenDialog = jest.fn()
    wrapper.instance().handleCloseDialog = jest.fn()
    wrapper.instance().handleToggleActive = jest.fn()

    const serviceComponent = wrapper.instance().renderService(args)

    serviceComponent.props.onEdit()

    expect(wrapper.instance().handleEdit).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleEdit).toHaveBeenCalledWith(args.item)

    serviceComponent.props.onDelete()

    expect(wrapper.instance().handleEdit).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleDelete).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleDelete).toHaveBeenCalledWith(args.item._id)

    serviceComponent.props.onOpenDialog()

    expect(wrapper.instance().handleEdit).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleDelete).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleOpenDialog).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleOpenDialog).toHaveBeenCalledWith(
      args.item._id,
    )

    serviceComponent.props.onCloseDialog()

    expect(wrapper.instance().handleEdit).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleDelete).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleOpenDialog).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleCloseDialog).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleCloseDialog).toHaveBeenCalledWith(
      args.item._id,
    )

    serviceComponent.props.onChangeStatus()

    expect(wrapper.instance().handleEdit).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleDelete).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleOpenDialog).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleCloseDialog).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleToggleActive).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleToggleActive).toHaveBeenCalledWith(
      args.item._id,
    )
  })
})
