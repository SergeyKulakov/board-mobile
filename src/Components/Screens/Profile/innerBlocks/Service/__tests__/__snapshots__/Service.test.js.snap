// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.Service render snapshot 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <ShadowBox>
    <Component />
    <Styled(View)>
      <FlatList
        data={Array []}
        disableVirtualization={false}
        extraData={
          Array [
            null,
            Object {
              "servicesLengthError": "servicesLengthError",
            },
          ]
        }
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </Styled(View)>
    <Styled(View)>
      <Button
        onClick={[Function]}
        style={
          Object {
            "color": "#69696a",
            "gradient": Object {
              "end": "#cbcbcb",
              "start": "#f1f1f1",
            },
          }
        }
      />
    </Styled(View)>
    <Styled(View)>
      <InputBlock
        animatedLabel={true}
        animatedLine={true}
        autoHeight={true}
        multiline={true}
        numberOfLines={1}
        onBlur={null}
        onChange={[Function]}
        testID="aboutInput"
      />
    </Styled(View)>
  </ShadowBox>
</TouchableWithoutFeedback>
`;
